File: web/src/types/logbook.ts
import type { Yacht } from "./yacht";
import type { Owner } from "./owner";

export type LogbookEntryType =
	| "departure"
	| "return"
	| "fuel"
	| "maintenance"
	| "general"
	| "incident";

export interface Booking {
	id: string;
	yacht_id: string;
	user_id: string;
	start_date: string;
	end_date: string;
	status: "pending" | "confirmed" | "cancelled" | "completed";
	booking_type: "regular" | "standby";
	is_standby: boolean;
	notes?: string;
	cancelled_at?: string;
	created_at: string;
	updated_at: string;
}

export interface LogbookEntry {
	id: string;
	yacht_id: string;
	booking_id?: string;
	user_id: string;
	entry_type: LogbookEntryType;
	fuel_liters?: number;
	fuel_cost?: number;
	hours_operated?: number;
	notes?: string;
	created_at: string;
	yacht?: Yacht;
	booking?: Booking;
	user?: Owner;
}

export interface CreateLogbookEntryRequest {
	yacht_id: string;
	entry_type?: LogbookEntryType;
	fuel_liters?: number;
	fuel_cost?: number;
	hours_operated?: number;
	notes?: string;
}

File: backend/internal/models/logbook.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type LogbookEntryType string

const (
	EntryTypeDeparture   LogbookEntryType = "departure"
	EntryTypeReturn      LogbookEntryType = "return"
	EntryTypeFuel        LogbookEntryType = "fuel"
	EntryTypeMaintenance LogbookEntryType = "maintenance"
	EntryTypeGeneral     LogbookEntryType = "general"
	EntryTypeIncident    LogbookEntryType = "incident"
)

const (
	TripEntryTypeDepart  LogbookEntryType = "departure"
	TripEntryTypeReturn  LogbookEntryType = "return"
)

type LogbookEntry struct {
	ID                   uuid.UUID        `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	YachtID              uuid.UUID        `gorm:"type:uuid;not null;index" json:"yacht_id"`
	BookingID            *uuid.UUID       `gorm:"type:uuid;index" json:"booking_id,omitempty"`
	UserID               uuid.UUID        `gorm:"type:uuid;not null;index" json:"user_id"`
	EntryType            LogbookEntryType `gorm:"type:varchar(20);not null;index" json:"entry_type"`
	PortEngineHours      *float64         `gorm:"type:decimal(10,2)" json:"port_engine_hours,omitempty"`
	StarboardEngineHours *float64         `gorm:"type:decimal(10,2)" json:"starboard_engine_hours,omitempty"`
	FuelLiters           *float64         `gorm:"type:decimal(10,2)" json:"fuel_liters,omitempty"`
	Notes                string           `gorm:"type:text" json:"notes,omitempty"`
	CreatedAt            time.Time        `gorm:"index" json:"created_at"`

	// Relationships
	Yacht   Yacht    `gorm:"foreignKey:YachtID;constraint:OnDelete:CASCADE" json:"yacht,omitempty"`
	Booking *Booking `gorm:"foreignKey:BookingID" json:"booking,omitempty"`
	User    User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

func (LogbookEntry) TableName() string {
	return "logbook_entries"
}
